### ISSUE TITLE
Unable to build code model for ''. (No model selected)

### ISSUE THREAD

**Issue by sn50** [codecadwallader/codemaid#192]
I've seen the exception for the first time (doesn't mean it didn't happen). It appeared on the visual studio startup, on solution opening. I think it was trying to build Spade content (since I'm using only the Spade feature - it is very excellent feature, I thank you for it!, and for me it is the only useful feature of your Codemaid extension) from nothing - no file was selected at that time. I even think that the solution explorer was not loaded at all.

I'll add more information if I notice it again.
```
[CodeMaid Handled Exception 11:31:00 AM] Unable to build code model for '': System.Runtime.InteropServices.COMException (0x80004005): Error HRESULT E_FAIL has been returned from a call to a COM component.
   at EnvDTE.TextDocument.get_StartPoint()
   at SteveCadwallader.CodeMaid.Helpers.TextDocumentHelper.FindMatches(TextDocument textDocument, String patternString)
   at SteveCadwallader.CodeMaid.Model.CodeModelHelper.RetrieveCodeRegions(TextDocument textDocument)
   at SteveCadwallader.CodeMaid.Model.CodeModelBuilder.RetrieveAllCodeItems(Document document)
   at SteveCadwallader.CodeMaid.Model.CodeModelManager.BuildCodeItems(CodeModel codeModel)
```

VS Professional 2013, 12.0.40629.00 Update 5
CodeMaid 0.9.0

**Comment by codecadwallader** [comment]
Thanks for reporting the issue.  That error will happen if it's trying to understand a document, but the document is no longer available or doesn't have recognizable code elements.  It isn't necessarily harmful just a notification, but I have seen another issue recently #185 where it concerned a user so I will look into it a little further to see if I can make it less noisy.

**Comment by sn50** [comment]
Is it important to show it at all? It seems to me in this case it should be hidden.

**Comment by codecadwallader** [comment]
I'm inclined to agree.  It's being surfaced as an exception because there is an underlying one that caused the build model to fail to load - but since it's a known and expected condition (e.g. unsupported languages) it shouldn't be making noise and causing confusion.

**Comment by codecadwallader** [comment]
Now available in the v0.9.0.1-beta release:
https://github.com/codecadwallader/codemaid/releases/tag/v0.9.0.1-beta
